sys/src/9/port/devramfs.c

```diff
@@ -253,15 +350,14 @@ ramcreate(Chan* c, char *name, int omode, int perm)
 
 	file->sibling = parent->firstchild;
 	parent->firstchild = file;
-	qunlock(&ramlock);
-
 	mkqid(&c->qid, (uintptr_t)file, 0, file->perm & DMDIR ? QTDIR : 0);
 
-        poperror();
-
         c->offset = 0;
         c->mode = omode;
         c->flag |= COPEN;
+	qunlock(&ramlock);
```

Comment:
we can look later at whether holding this lock for this big chunk impacts performance.